### ISSUE TITLE
Avatar and stuff from other instance not fetched

### ISSUE THREAD

**Issue by R0ckweb** [tootsuite/mastodon#1141]
My two instances do not fetch avatar and stuff from other instance since rev 5442083 :-/
Users profil are empty, no avatar, no description, no banner.

Reset to c19e0f1 solve this issue.

* * * *

- [x ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by Gargron** [comment]
A good while ago a new type of sidekiq queue was added, now it's:

- default
- push
- pull
- mailers

Fetching and updating of profiles **has been moved to the pull queue**. This has various performance benefits of not holding down Sidekiq from processing higher priority tasks. Anyway, the docker-compose.yml and Procfile already contain the correction. Documentation has also been updated to include it in the systemd service. This should be an easy change. Let me know if that resolves your issue.

**Comment by nlienard** [comment]
i"ve same issue. i don't understand the last comment related to systemd service.
What should we do exactly ?
thanks

**Comment by R0ckweb** [comment]
I already updated sidekiq service :-/

![image](https://cloud.githubusercontent.com/assets/15029076/24799690/e5193c56-1b9c-11e7-9ad7-b2b44831217e.png)

**Comment by Spunkie** [comment]
Possible duplicate of https://github.com/tootsuite/mastodon/issues/1126

**Comment by kuro5hin** [comment]
I'm seeing long delays pulling some remote avatars and profile info. Some come in right away, some take hours, I don't see any pattern in which happens when (it's not by server, it doesn't seem to matter who's following them, etc). I'm running docker, rebuilt this morning form the current master.

**Comment by maximeborges** [comment]
Same problem on dedicated using Docker. I can't get the number of followers nor the number that one follows. Installation is from 2017-04-05, and I just rebuild all images 5 minutes ago.

**Comment by wirehack7** [comment]
I ran into the same problem. I'm using the latest version with docker. Sidekiq job is also working with pull. 
Avatars are only saved when I follow the user of the other instance (only user on instance). If not, no avatar of the user will be shown. Just the default one.
Also pulling infos is not really working. I don't see the hole followers and posts of other users of other instances.

**Comment by colegleason** [comment]
I'm also seeing issues with loading avatars, etc. If I follow someone, it seems to eventually load. Otherwise, I can't see remote avatars, follower info, etc.

**Comment by R0ckweb** [comment]
Maybe : 

`2017-04-07T20:18:48.749Z 8849 TID-orwndcn38 RemoteProfileUpdateWorker JID-40afdeb780928996453d000c INFO: start`
`Account Load (0.6ms)  SELECT  "accounts".* FROM "accounts" WHERE "accounts"."id" = $1 LIMIT $2  [["id", nil], ["LIMIT", 1]]`
`2017-04-07T20:18:48.751Z 8849 TID-orwndcn38 RemoteProfileUpdateWorker JID-40afdeb780928996453d000c INFO: done: 0.003 sec `

id nil ?

**Comment by rhaamo** [comment]
Same issue here, git commit 541c538f9bbd00028b388468b2753b1136f3d5b3 (latest ATM).
Also sidekiq ok, 4 queues running.
No visible errors except the same nil as @R0ckweb

**Comment by R0ckweb** [comment]
@rhaamo This PR https://github.com/tootsuite/mastodon/pull/1218 solve the problem 

Waiting to merge ;-)

**Comment by maximeborges** [comment]
So is it possible to regenerate all the accounts ? It seems that I still have lots of empty accounts

**Comment by rhaamo** [comment]
I'm still getting some issues here. git pull'ed and db:migrate && assets:precompile but still get empty profiles.
Click on user to display on sidebar doesn't load anything.
BUT, If I try to @ him in the toot-box, it instant-refresh when the auto-completion list pop.
Any thoughts ?

**Comment by kuro5hin** [comment]
It seems like every account that was already in my instance and suffered from this bug is still messed up -- no avatar, no bio. And they don't refresh themselves at all, for me -- I've tried following, replying, everything I could think of. Is there something we could run to fix the existing broken users now that the bug is patched?

**Comment by nlienard** [comment]
After the patch, my instance seems OK now, i got the avatar showing on Federal Feed !

**Comment by maximeborges** [comment]
Updated today, seems to be ok now

**Comment by guanix** [comment]
I am still experiencing this on latest master.

**Comment by R0ckweb** [comment]
@guanix like kuro5hin says, accounts that was already in your instance database will not be refresh.
Make a test on remote account that are you absolutely certain is not already in your database !

For exemple test remote follow on my account (no need to follow me)
1) Visit https://mastodon.land/users/Yann/remote_follow
2) Enter your account and click "Proceed to follow"
3) Close tab and wait 10 sec.(deppend on your instance load)
4) Visit again https://mastodon.land/users/Yann/remote_follow
5) Enter your accountand click "Proceed to follow" 
This time my account informations will be visible.

Of course this exemple work only if my account is not already know of your instance ;-)

**Comment by guanix** [comment]
@R0ckweb Thanks. I might have a different problem? I wasn’t completely sure about you, so I found a random person on Twitter who said he just signed up (jcape@mstdn.io), and followed those steps. I clicked “Follow” at the end of the process as step 6, but did not click “Follow” after step 2.

Here’s what was in the log as I did this. The weird thing is, it actually appears to download the avatar and run ImageMagick on it?

[Here’s a gist of what the log said](https://gist.github.com/guanix/2da9c05ce891df8210881e081c3315e0). The URL where this user’s avatar is supposed to end up is [here](https://nsfw.finance/system/accounts/avatars/000/000/044/original/e306225381720abd.jpg?1491782673). I am running 93db265be7b648fe095d5a92b76c5c7077c72ac2. Local avatars all display correctly, so it doesn’t seem to be a problem relating to permissions for the `system` folder or anything like that.
